### DISCUSSION TITLE
Understanding HMI

### DISCUSSION THREAD

**Discussion by 13Aman** [ApolloAuto/apollo#15340] [Q&A]
I am looking at hmi section code in apollo/modules/dreamview/backend/hmi/.
Let me share what I have understood from this code. This hmi code is a part of dreamview module. Dreamview is a web application, which shows information about the autonomous vehicle, like future trajectory, vehicle localization etc on the screen. HMI in particular shares information with the dreamview which shows it on the screen. It shows information like audio, drive event, other HMI actions like enter auto mode load records etc.

Please tell if I have understood it correctly, and what is that I am lacking. Please tell something so that I can enhance my knowledge

**Comment by daohu527** [comment]
Dreamview adopts a front-end and back-end model. The front-end is responsible for display and the back-end is responsible for processing logic. They communicate through **websocket**.
